proto/nbd.proto

```diff
@@ -0,0 +1,77 @@
+syntax = "proto3";
+
+package nbd;
+
+// Remote block device service.
+//
+// Clients can use this service to read and write blocks from a remote device
+// that serves the underlying block data.
+//
+// Servers are responsible for writing and serving the block data to persistent
+// storage.
+service BlockDevice {
+  // Returns metadata for a device label. The label is a well-known value by
+  // both the client and server that corresponds to a desired device.
+  rpc Metadata(MetadataRequest) returns (MetadataResponse);
+
+  // Reads a block from a device with a given ID.
+  rpc Read(ReadRequest) returns (ReadResponse);
+
+  // Writes a block to a device with a given ID.
+  rpc Write(WriteRequest) returns (WriteResponse);
+}
+
+message DeviceMetadata {
+  // The device ID to be included in IO requests.
+  uint64 device_id = 1;
+
+  // The device label used to identify the device in metadata requests.
+  // Ex: "scratchfs"
+  string label = 2;
+
+  // The size of the block device in bytes.
+  uint64 size_bytes = 3;
```

Comment:
int64

Reply:
Obvi don't have a good style guide link here (goog internal only), but basically you don't want to make values like this unsigned just because you don't think they'll be negative, because you may later do math on them and get a weird result.

The read/write APIs in go are going to deal in either int64 or int anyway, so you will also probably reduce conversions this way.

https://jacobegner.blogspot.com/2019/11/unsigned-integers-are-dangerous.html

Reply:
Nice, I hadn't thought of that before. Done!